numba/cpython/setobj.py

```diff
@@ -1328,21 +1350,54 @@ def set_remove(context, builder, sig, args):
     return context.get_dummy_value()
 
 
+@intrinsic
+def _set_remove(typingctx, s, item):
+    sig = types.none(s, item)
+    return sig, set_remove
+
+
+@overload_method(types.Set, "remove")
+def ol_set_remove(s, item):
+    # XXX: should item have the same type as s.dtype?
```

Comment:
I think yes given the current implementation. e.g.:
```python
from numba import njit
import numpy as np

@njit
def foo():
    x = {1, 2, 3, 4}
    x.remove('a')

foo()
```
ends up with:
```
TypeError: Type of #3 arg mismatch: i64 != {i8*, i64, i32, i32, i64, i8*, i8*}
```
As the reflected set implementation is pending deprecation I don't think there's any need to add support for this, but the issue in lowering can be prevented at type inference time so it should be. Suggest fixing the typing and adding a test.

Reply:
Thanks for making the change, I think it would be good to add a test for it.

Reply:
Oops, sorry. I forgot to include a test for it